js/modules/k6/http/http.go

```diff
@@ -21,101 +21,152 @@
+	mustExport("file", mi.File) // TODO: deprecate or refactor?
+
+	// TODO: refactor so the Client actually has better APIs and these are
+	// wrappers (facades) that convert the old k6 idiosyncratic APIs to the new
+	// proper Client ones that accept Request objects and don't suck
+	mustExport("get", func(url goja.Value, args ...goja.Value) (*Response, error) {
+		args = append([]goja.Value{goja.Undefined()}, args...) // sigh
+		return mi.defaultClient.Request(http.MethodGet, url, args...)
+	})
+	mustExport("head", mi.defaultClient.getMethodClosure(http.MethodHead))
+	mustExport("post", mi.defaultClient.getMethodClosure(http.MethodPost))
+	mustExport("put", mi.defaultClient.getMethodClosure(http.MethodPut))
+	mustExport("patch", mi.defaultClient.getMethodClosure(http.MethodPatch))
+	mustExport("del", mi.defaultClient.getMethodClosure(http.MethodDelete))
+	mustExport("options", mi.defaultClient.getMethodClosure(http.MethodOptions))
+	mustExport("request", mi.defaultClient.Request)
+	mustExport("batch", mi.defaultClient.Batch)
+	mustExport("setResponseCallback", mi.defaultClient.SetResponseCallback)
+
+	mustExport("expectedStatuses", mi.expectedStatuses) // TODO: refactor?
+
+	// TODO: actually expose the default client as k6/http.defaultClient when we
+	// have a better HTTP API (e.g. proper Client constructor, an actual Request
+	// object, custom Transport implementations you can pass the Client, etc.)
+
+	return mi
 }
 
-//nolint: golint
-type HTTP struct {
-	TLS_1_0                            string `js:"TLS_1_0"`
-	TLS_1_1                            string `js:"TLS_1_1"`
-	TLS_1_2                            string `js:"TLS_1_2"`
-	TLS_1_3                            string `js:"TLS_1_3"`
-	OCSP_STATUS_GOOD                   string `js:"OCSP_STATUS_GOOD"`
-	OCSP_STATUS_REVOKED                string `js:"OCSP_STATUS_REVOKED"`
-	OCSP_STATUS_SERVER_FAILED          string `js:"OCSP_STATUS_SERVER_FAILED"`
-	OCSP_STATUS_UNKNOWN                string `js:"OCSP_STATUS_UNKNOWN"`
-	OCSP_REASON_UNSPECIFIED            string `js:"OCSP_REASON_UNSPECIFIED"`
-	OCSP_REASON_KEY_COMPROMISE         string `js:"OCSP_REASON_KEY_COMPROMISE"`
-	OCSP_REASON_CA_COMPROMISE          string `js:"OCSP_REASON_CA_COMPROMISE"`
-	OCSP_REASON_AFFILIATION_CHANGED    string `js:"OCSP_REASON_AFFILIATION_CHANGED"`
-	OCSP_REASON_SUPERSEDED             string `js:"OCSP_REASON_SUPERSEDED"`
-	OCSP_REASON_CESSATION_OF_OPERATION string `js:"OCSP_REASON_CESSATION_OF_OPERATION"`
-	OCSP_REASON_CERTIFICATE_HOLD       string `js:"OCSP_REASON_CERTIFICATE_HOLD"`
-	OCSP_REASON_REMOVE_FROM_CRL        string `js:"OCSP_REASON_REMOVE_FROM_CRL"`
-	OCSP_REASON_PRIVILEGE_WITHDRAWN    string `js:"OCSP_REASON_PRIVILEGE_WITHDRAWN"`
-	OCSP_REASON_AA_COMPROMISE          string `js:"OCSP_REASON_AA_COMPROMISE"`
+func (mi *ModuleInstance) Exports() modules.Exports {
+	return modules.Exports{
+		Default: mi.exports,
+		// TODO: add new HTTP APIs like Client, Request, etc. as named exports?
```

Comment:
see https://github.com/grafana/k6/issues/2258 as it was more or less written for this module :rofl:

Reply:
I started by reading that issue, but I am still not sure exactly what you mean, sorry :disappointed: Do you mean I should just return everything as named exports here and rely on this? https://github.com/grafana/k6/blob/5c14dae129a5eb6ed309c94a194370878b7b08e5/js/initcontext.go#L180-L182